dns.go

```diff
@@ -80,6 +92,23 @@ func (c *Client) ListDomains() ([]Domain, error) {
 	return ds, nil
 }
 
+// NewDomain registers a new Domain
+func (c *Client) NewDomain(name string) (*Domain, error) {
```

Comment:
Could you make this `CreateDomain` please. It's always felt like `NewObject` should create a new object, but maybe not persist it, but `Create` actually saves it.